hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -168,6 +168,7 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
       if (purge && lastMetadata != null) {
         CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        CatalogUtil.dropTableBaseLocation(ops.io(), lastMetadata.location());
```

Comment:
I don't think this should be the default behavior. It is fine if people want to do this in catalogs, but I would prefer to have an option to turn it on or off.